pretyping/evarconv.ml

```diff
@@ -1046,8 +1098,15 @@ and evar_eqappr_x ?(rhs_is_already_stuck = false) flags env evd pbty
           (try
              if not flags.with_cs then raise Not_found
              else conv_record flags env
-               (try check_conv_record env i appr1 appr2
-                with Not_found -> check_conv_record env i appr2 appr1)
+               (let x = try get_cs env i keys2 (match lastUnfolded with | Some true -> true | _ -> false) parents2 appr1 c2 appr2
+                 with No_cs b -> no_cs1 := b; None in
+               let x = match x with
+               | Some _ -> x
```

Comment:
I think you should name the `try...with No_cs ... ` code, maybe `get_cs_opt` and move out the assignment to no_cs1. This way you should be able to call it twice and the control flow should  become a bit less clear as in
```
match get_cs_opt bla bla 2 with
| Some x -> x
| None ->
   match get_cs_opt bla bla 1 with
  | Some x -> x
  | None -> raise Not_found
```